client/asset/btc/rpcclient.go

```diff
@@ -495,11 +495,35 @@ func (wc *rpcClient) listUnspent() ([]*ListUnspentResult, error) {
 	return unspents, wc.call(methodListUnspent, anylist{uint8(0)}, &unspents)
 }
 
+// lockAllUnspent unlocks all locked outputs for spending again. If unlock is
+// true btc will return 'true' if successful. If unlock is false btc will also
+// return 'true' but do nothing so a warning is issued below.
+//
+// See Also: btc_test.go: RawRequest case: methodLockUnspent
+func (wc *rpcClient) lockAllUnspent(unlock bool) error {
+	if !unlock {
+		wc.log.Warn("attempting to lock all unspent outputs 'lockunspent false'")
+	}
+	var rpcReturn bool
+	err := wc.call(methodLockUnspent, anylist{unlock}, &rpcReturn)
+	if err != nil {
+		return fmt.Errorf("'lockunspent %v': %v: ", unlock, err)
+	}
+	if !rpcReturn {
+		return fmt.Errorf("'lockunspent %v' unsuccessful: RPC returns %v",
+			unlock, rpcReturn)
+	}
+	return nil
+}
+
 // lockUnspent locks and unlocks outputs for spending. An output that is part of
 // an order, but not yet spent, should be locked until spent or until the order
 // is canceled or fails.
 func (wc *rpcClient) lockUnspent(unlock bool, ops []*output) error {
-	var rpcops []*RPCOutpoint // To clear all, this must be nil->null, not empty slice.
+	if ops == nil {
```

Comment:
comparing a slice to nil is often error prone.  Reason not to check `len(ops) == 0`?

Reply:
Perhaps inexperience with golang.

Reply:
There is probably more wrong with `ReturnCoins(nil)` anyway. It was first written against btc. But some older clones do not seem to work quite the same way. 

I see for at least one coin that even the more correct  `lockunspent true` does not work as I would expect. I sometimes see, for example, 3 coins locked -- then `lockunspent true` but instead of individual coins returned to unspent they are amalgamated into one - sometimes - hard to catch.

My current thinking is to Not use `lockunspent true` at all for all coins. Just take the less efficient path of iterating each coin and unlocking individually in a loop. It works for all clones as far as I can see from my testing some weeks ago; and is probably the more tested C++ code path.

We currently do not use the `ReturnCoins(nil)` version of the function much. I did a search before and the results are [above](https://github.com/decred/dcrdex/pull/2397#issuecomment-1594887959)

Switching to (finding then) iterating over locked coins seems non-breaking, then. I would leave the `btc_test.go` as is.

What do you think?